Lib/test/_test_multiprocessing.py

```diff
@@ -3610,6 +3616,234 @@ def test_copy(self):
         self.assertAlmostEqual(bar.y, 5.0)
         self.assertEqual(bar.z, 2 ** 33)
 
+
+@unittest.skipUnless(HAS_SHMEM, "requires multiprocessing.shared_memory")
+class _TestSharedMemory(BaseTestCase):
```

Comment:
is there a reason for this to be this private? (underscore)

Reply:
It seems this is class is not executed. Looks like you want to inherit from unittest,TestCase.

Reply:
It is executing -- I see it in the tests being run when I enable `-v`.  I see it running on the buildbots too (more reassuring to me).

The pattern used by this class is being used on a number of others and is motivated/related to the controls for when a test runs under fork/forkserver/spawn.

Reply:
It doesn't run with:

```
./python -m test _test_multiprocessing -v -m _TestSharedMemory
```

Reply:
Correct.  Instead something dynamic happens to the name.  Check out what happens to the name:
```
test_shared_memory_ShareableList_basics (test.test_multiprocessing_spawn.WithProcessesTestSharedMemory) ... ok
test_shared_memory_ShareableList_pickling (test.test_multiprocessing_spawn.WithProcessesTestSharedMemory) ... ok
test_shared_memory_SharedMemoryManager_basics (test.test_multiprocessing_spawn.WithProcessesTestSharedMemory) ... ok
test_shared_memory_across_processes (test.test_multiprocessing_spawn.WithProcessesTestSharedMemory) ... ok
test_shared_memory_basics (test.test_multiprocessing_spawn.WithProcessesTestSharedMemory) ... ok
```

This is due to ``ALLOWED_TYPES = ('processes',)`` appearing at the head of the test class.

Reply:
It appears `BaseTestCase` does not inherit from `unittest.TestCase`, that's why it's not possible to run this test individually.